source/Cosmos.HAL2/PIT.cs

```diff
@@ -167,27 +178,36 @@ public class PITTimer : IDisposable {
             }
         }
 
-        public void HandleInterrupt() {
-            int T0Delay = (int)T0DelyNS;
-            PITTimer hndlr = null;
-            for (int i = ActiveHandlers.Count - 1; i >= 0; i--) {
-                hndlr = ActiveHandlers[i];
+	    private void HandleIRQ(ref INTs.IRQContext aContext)
+	    {
 
-                hndlr.NSRemaining -= T0Delay;
+	        int T0Delay = (int) T0DelyNS;
+	        PITTimer hndlr = null;
+	        for (int i = ActiveHandlers.Count - 1; i >= 0; i--)
```

Comment:
Add a line before the for loop to reset the countdown after the irq fires.

T0Countdown = 65535;